jerry-ext/private-headers/section-utils.h

```diff
@@ -0,0 +1,67 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef SECTION_UTILS_H
+#define SECTION_UTILS_H
+
+/**
+ * Declare references to a section that contains an array of items.
+ *
+ * @name the name of the section (without quotes)
+ * @type the type of the elements stored in the array
+ *
+ * This macro needs to be placed before any occurrence of ITERATE_OVER_SECTION, for a given section.
+ */
+#ifdef __MACH__
+#define DECLARE_SECTION(name, type)                                  \
+extern type __start_ ## name[] __asm("section$start$__DATA$" #name); \
+extern type __stop_ ## name[] __asm("section$end$__DATA$" #name);
+#else /* !__MACH__ */
+#define DECLARE_SECTION(name, type)                   \
+extern type __start_ ## name[] __attribute__((weak)); \
+extern type __stop_ ## name[] __attribute__((weak));
```

Comment:
Even for internal usage like this, it's still a good idea, because if we need to
```C
#ifdef PLATFORM_SUPPORTS_SECTIONS
#define DECLARE_SECTION(name, type) /* leaves semicolon to callers */
#else /* !PLATFORM_SUPPORTS_SECTIONS */
#define DECLARE_SECTION(name, type) \
  void something(void) __attribute__((constructor)) {}
#endif /* PLATFORM_SUPPORTS_SECTIONS */
```
Then it's best to not also have to add an ifdef to the usage like so:
```
#ifdef PLATFORM_SUPPORTS_SECTIONS
DECLARE_SECTION(jerryx_ctx_slots, jerryx_context_slot_t);
#else
DECLARE_SECTION(jerryx_ctx_slots, jerryx_context_slot_t)
#endif
```